Source/WebCore/animation/ViewTimeline.cpp

```diff
@@ -157,20 +157,29 @@ void ViewTimeline::cacheCurrentTime()
     m_cachedCurrentTimeData = [&] -> CurrentTimeData {
         if (!m_subject)
             return { };
+
         CheckedPtr subjectRenderer = m_subject->renderer();
         if (!subjectRenderer)
             return { };
-        auto* sourceScrollableArea = scrollableAreaForSourceRenderer(sourceScrollerRenderer(), m_subject->document());
+
+        CheckedPtr sourceRenderer = sourceScrollerRenderer();
+        auto* sourceScrollableArea = scrollableAreaForSourceRenderer(sourceRenderer.get(), m_subject->document());
         if (!sourceScrollableArea)
             return { };
+
         auto scrollDirection = resolvedScrollDirection();
         if (!scrollDirection)
             return { };
 
         float scrollOffset = scrollDirection->isVertical ? sourceScrollableArea->scrollOffset().y() : sourceScrollableArea->scrollOffset().x();
         float scrollContainerSize = scrollDirection->isVertical ? sourceScrollableArea->visibleHeight() : sourceScrollableArea->visibleWidth();
-        auto subjectOffsetFromSource = subjectRenderer->localToContainerPoint(pointForLocalToContainer(*sourceScrollableArea), sourceScrollerRenderer());
+        auto subjectOffsetFromSource = subjectRenderer->localToContainerPoint(pointForLocalToContainer(*sourceScrollableArea), sourceRenderer.get());
         float subjectOffset = scrollDirection->isVertical ? subjectOffsetFromSource.y() : subjectOffsetFromSource.x();
+
+        // Ensure borders are subtracted.
+        auto scrollerPaddingBoxOrigin = sourceRenderer->paddingBoxRect().location();
+        subjectOffset -= scrollDirection->isVertical ? scrollerPaddingBoxOrigin.y() : scrollerPaddingBoxOrigin.x();
```

Comment:
Might have to think about writing modes here.

Reply:
The `scrollDirection` is aware of writing modes, see https://github.com/WebKit/WebKit/commit/8ddc6659d2b26ee20a772dc838a3e5c37ae94ab5.